### ISSUE TITLE
Qt Creator 4.12 build problems

### ISSUE THREAD

**Issue by NickLion** [hugopl/RubyCreator#131]
Hi,

There have been released a new version of Qt Creator (4.12) and RubyCreator doesn't build against it. Have you looked into it already or you haven't had time for that yet? I'm not familiar with Qt Creator plugin development, so I may look at it on weekend to fix, but if you have already started, that would be definitely faster.

**Comment by hugopl** [comment]
I'm aware of this problem, but I didn't had time to work on it and figure out all the API changes in QtC, to be honest, keep a plugin compiling for QtC is a hell, they change the API **a lot** on every minor version and they don't even document what changed, the last couple compile fixes were done by @orgads (kudos to him) who is also a QtC dev.

I tried to fix the compilation yesterday but noticed that it would require some time digging the QtC commits to figure out what changed.

As a temporary fix, on ArchLinux I told pacman to not upgrade qtcreator package, and installed it ignoring the missing dependency on clang 9.0, as nowadays I don't do paid-work anymore with C++ it wasn't a problem for me, but may be for others.

TL;DR; I'm aware... tired of QtC API changes but trying to fix, since I use this plugin myself every working day, and thanks for reporting.

**Comment by hugopl** [comment]
I created a branch named `4.12.x-wip` while I'm slowly fixing the compile errors... when it's done I merge it into master and tag it.

**Comment by NickLion** [comment]
Thanks, I've noticed that some functions moved to BuildSystem, but code is very new to me. So right now it builds but doesn't work "fine". For me it loads the project but files has path in their names and no signs of rubocop or navigation. And highlighting is done by KF. Is this some my build problem?

**Comment by NickLion** [comment]
Hm, I've reopened files and seems rubocop, highlighting and navigation work fine. But files still have path in their names in project explorer. It looks like
```
app
  controller
    app/controller/application_controller.rb
```

**Comment by orgads** [comment]
I'll try to have a look soon. Can't promise when though :)

**Comment by NickLion** [comment]
@orgads thanks, this doesn't bother me too much as it doesn't stop from work, so no hurry.

**Comment by hugopl** [comment]
I saw this just now... usually I keep the project tree hidden.

**Comment by hugopl** [comment]
fixed here... sending a path soon...

**Comment by hugopl** [comment]
Fixed in 7151bf1a2b08797ee64f2e7987b707c4579fa238.

@orgads can I add you to the project? So you can review my patches as well. But if you are fine the way it's right now it's ok too.

**Comment by orgads** [comment]
Sure, no prob :)

**Comment by NickLion** [comment]
Thanks!
